src/mcts/search.cc

```diff
@@ -1980,45 +1980,14 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
       }
       return NodeToProcess::Collision(node, depth, collision_limit);
     }
-    // If terminal, we either found a twofold draw to be reverted, or
-    // reached the end of this playout.
+
+    // Probably best place to check for two-fold draws consistently.
+    // Depth starts with 1 at root, so real depth is depth - 1.
+    EnsureNodeTwoFoldCorrectForDepth(node, depth - 1);
```

Comment:
Any particular reason this was moved before the if?

Reply:
Because if its inside the if - we have to check IsTerminal again afterwards, since if we revert two fold it shouldn't count as a visit. (Previously that path fell through - but as this is a standalone function it isn't a straight forward transform.)
Moving it outside the if is safe because IsTwoFoldTerminal (checked inside the function) can only be true if IsTerminal is also true.

Reply:
(I also assume the compiler will inline/partial inline as appropriate.)